org/oppia/proto/v1/api/android.proto

```diff
@@ -0,0 +1,391 @@
+  org.oppia.proto.v1.versions.LanguageProtosVersion language_protos_version = 9;
+
+  // The latest proto version for images that the client can support.
+  org.oppia.proto.v1.versions.ImageProtoVersion thumbnail_proto_version = 10;
+}
+
+// Details about a versione dstructure that's compatible with the local client. This can be used
+// both to indicate the local download state of a client's topics, and to provide details for
+// available changes to client-compatible topics.
+//
+// Note also that all structures represented by this message are considered "global" (that is, not
+// tied to any particular topics) even though this message may be used in the context of a topic
+// (generally to represent structures corresponding to the learning experience provided by that
+// topic). This is an important distinction since some structures (like concept cards) aren't
+// actually tied to any particular topics, and can be used across multiple topics. However, they are
+// downloaded with the first topic that references them, and shouldn't be re-downloaded in
+// subsequent topics (the backend makes guarantees to ensure that structures are never repeated).
+message VersionedStructureDetailsDto {
+  option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+  // The backend-tracked and creator-affected content version of the structure. This is the primary
+  // indicator of how "new" a particular structure is, or whether it has updates available. This is
+  // always a monotonically increasing positive integer for newer versions.
+  uint32 content_version = 1;
+
+  // The type of structure being detailed.
+  oneof structure_type {
+    // Indicates that these details correspond to a complete topic, as specified by its ID.
+    string topic_id = 2;
+
+    // Indicates that these details correspond to a topic summary, as specified by a topic ID.
+    string topic_summary_id = 3;
+
+    // Indicates that these details correspond to a revision card, as specified by its ID structure.
+    org.oppia.proto.v1.structure.SubtopicIdDto revision_card_id = 4;
+
+    // Indicates that these details correspond to a concept card, as specified by its skill ID.
+    string concept_card_id = 5;
+
+    // Indicates that these details correspond to a story, as specified by its ID.
+    string story_id = 6;
+
+    // Indicates that these details correspond to an exploration, as specified by its ID.
+    string exploration_id = 7;
+
+    // Indicates that these details correspond to a question, as specified by its ID.
+    string question_id = 8;
+
+    // Indicates that these details correspond to a skill, as specified by its ID.
+    string skill_id = 9;
+  }
+}
+
+// An identifier used to download a specific structure at a particular content version as part of
+// the topic content API. It's expected that the server-provided structure will utilize the provided
```

Comment:
Probably strike the "provided" at the end of this line, I think it's not needed (it's inherent in "requested").